frontend/src/utils/helpers/moduleUtils.js

```diff
@@ -14,12 +14,15 @@ const ModuleNames = {
 };
 
 function isModuleEnabled(module) {
-  if (module === ModuleNames.CATALOG) {
```

Comment:
This is good for V2.0.0 but maybe we can already propose abstractions to handle dependencies that are more generic, similar to the `depends_on` method in the backend. wdyt?

Reply:
could you please elaborate? I'm not quite familiar with backend.

Reply:
In this `utils` we are defining that `Shares` depends on `Datasets` and that `Catalog` and `Glossaries` depend on `Datasets` and `Dashboards`. I find it strange that the dependencies of the module are defined in the utils and not in the module itself. For me `isModuleEnabled` should check the dependencies with something like getModuleDependencies(module) and then return getModuleActiveStatus(dependency). An abstraction of the dependencies instead of hardcoding Datasets, Dashboards....

Reply:
In the backend the depedency is defined in the `__init__` of each module, I am not sure what would be the equivalent in frontend

Reply:
Make perfect sense, I'll open an issue with the proposed suggestion.